### PR TITLE
clean up codes after PodDisruptionConditions was promoted to GA 

### PR DESCRIPTION
**Pull request by carlory**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

A follow-up of https://github.com/kubernetes/kubernetes/pull/125461

#### Which issue(s) this PR fixes:

Ref https://github.com/kubernetes/kubernetes/pull/125461

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs
- [KEP]: https://github.com/kubernetes/enhancements/tree/master/keps/sig-apps/3329-retriable-and-non-retriable-failures
```

### CODE REVIEW

File: pkg/controller/podgc/gc_controller.go [hunk 1]
```diff
@@ -343,24 +341,20 @@ func (gcc *PodGCController) markFailedAndDeletePodWithCondition(ctx context.Cont
...
+		newStatus := pod.Status.DeepCopy()
+		newStatus.Phase = v1.PodFailed
+		if condition != nil {
+			apipod.UpdatePodCondition(newStatus, condition)
+		}
+		if _, _, _, err := utilpod.PatchPodStatus(ctx, gcc.kubeClient, pod.Namespace, pod.Name, pod.UID, pod.Status, *newStatus); err != nil {
+			return err
 		}
 	}
+
```
Comment (mimowo) [inline pkg/controller/podgc/gc_controller.go:358]: nit: cleanup the formatting change

File: pkg/controller/podgc/gc_controller.go [hunk 2]
```diff
@@ -343,24 +341,20 @@ func (gcc *PodGCController) markFailedAndDeletePodWithCondition(ctx context.Cont
 	logger := klog.FromContext(ctx)
 	logger.Info("PodGC is force deleting Pod", "pod", klog.KObj(pod))
 	// Patch the pod to make sure it is transitioned to the Failed phase before deletion.
-	// This is needed for the JobPodReplacementPolicy feature to make sure Job replacement pods are created.
-	// See https://github.com/kubernetes/enhancements/tree/master/keps/sig-apps/3939-allow-replacement-when-fully-terminated#risks-and-mitigations
-	// for more details.
-	if utilfeature.DefaultFeatureGate.Enabled(features.PodDisruptionConditions) || utilfeature.DefaultFeatureGate.Enabled(features.JobPodReplacementPolicy) {
```
Comment (mimowo) [inline pkg/controller/podgc/gc_controller.go:349]: With this change we can drop `JobPodReplacementPolicy` from the test: https://github.com/kubernetes/kubernetes/blob/0c8b3e5f305bf2bf56d47019199b81330d90c2c3/test/integration/podgc/podgc_test.go#L58

**Review by mimowo — COMMENTED** [review]
Thanks, lgtm, just nits

File: pkg/controller/podgc/gc_controller.go (hunk 2 above)
Comment (carlory) [reply pkg/controller/podgc/gc_controller.go:349]: removed

**Review by mimowo — COMMENTED** [review]
/lgtm
/cc @SergeyKanzhelev @soltysh

**Comment by carlory** [comment]
@mimowo Fixed unit test, please review again. thanks.

**Review by mimowo — COMMENTED** [review]
/lgtm

**Comment by mimowo** [comment]
/test pull-kubernetes-integration
Overloaded infra: "There are no nodes that your pod can schedule to - check your requests, tolerations, and node selectors"

File: pkg/controller/job/job_controller.go [hunk 3]
```diff
@@ -1875,9 +1875,8 @@ func ensureJobConditionStatus(list []batch.JobCondition, cType batch.JobConditio
 }
 
 func isPodFailed(p *v1.Pod, job *batch.Job) bool {
-	if feature.DefaultFeatureGate.Enabled(features.PodDisruptionConditions) && feature.DefaultFeatureGate.Enabled(features.JobPodFailurePolicy) && job.Spec.PodFailurePolicy != nil {
-		// When PodDisruptionConditions is enabled, orphan Pods and unschedulable
-		// terminating Pods are marked as Failed. So we only need to check the phase.
+	if feature.DefaultFeatureGate.Enabled(features.JobPodFailurePolicy) && job.Spec.PodFailurePolicy != nil {
+		// Orphan Pods and unschedulable terminating Pods are marked as Failed. So we only need to check the phase.
 		// TODO(#113855): Stop limiting this behavior to Jobs with podFailurePolicy.
```
[the remark this answers is not in the document]
Comment (alculquicondor) [reply pkg/controller/job/job_controller.go:1880]: or would we not be able to implement the Pod replacement policy TerminatingOrFailed if we always used this check?

[the remark this answers is not in the document]
Comment (alculquicondor) [reply pkg/controller/job/job_controller.go:1880]: oh, #125997 handles this. Then this PR will have to rebase.

[the remark this answers is not in the document]
Comment (alculquicondor) [reply pkg/controller/job/job_controller.go:1880]: I'm ok keeping the existing behavior for the long term. The fact that it applies to jobs using failure policies sounds good enough.

**Comment by carlory** [comment]
/hold cancel

rebased.

/assign @derekwaynecarr @kow3ns @mikedanese
/cc @mimowo

**Review by soltysh — APPROVED** [review]
/lgtm
/approve
for sig-apps bits

**Comment by mimowo** [comment]
/assign @SergeyKanzhelev 
for the sig-node bits
/cc @smarterclayton

**Review by SergeyKanzhelev — APPROVED** [review]
/lgtm

**Pull request merged**